enterprise/internal/insights/resolvers/insight_view_resolvers.go

```diff
@@ -99,10 +99,18 @@ func (i *insightViewResolver) DataSeries(ctx context.Context) ([]graphqlbackend.
 	return resolvers, nil
 }
 
-func (i *insightViewResolver) Presentation(ctx context.Context) (graphqlbackend.InsightPresentation, error) {
-	lineChartPresentation := &lineChartInsightViewPresentation{view: i.view}
+func (i *insightViewResolver) PresentationType(ctx context.Context) (string, error) {
+	return i.view.PresentationType, nil
+}
 
-	return &insightPresentationUnionResolver{resolver: lineChartPresentation}, nil
+func (i *insightViewResolver) Presentation(ctx context.Context) (graphqlbackend.InsightPresentation, error) {
+	if i.view.PresentationType == "PIE" {
```

Comment:
For more type safety, we could also have the type on `Insight` be `PresentationType` and you should be able to scan that directly as a string. Then you should be able to compare with your const values like below:
```suggestion
	if i.view.PresentationType == types.Pie {
```

Reply:
Ah yeah thank you! I think I wrote this before I created the enum and forgot to go back and change it. 👍